coursemapper-kg/recommendation/app/services/course_materials/db/neo4_db.py

```
@@ -871,76 +871,72 @@ def get_or_create_resoures_relationships(self, user_id,material_id, concept_ids,
-                resources.extend([r for r in resource if r not in resources])
-                relationships.extend([r for r in relationship if r not in relationships])
+        with self.driver.session() as session:
+            try:
+                if len(wikipedia_articles) != 0:
+                    for index, resource in wikipedia_articles.iterrows():
+                        rid = resource['id']
+                        weight = _truncate(resource[recommendation_type] if recommendation_type in resource.index else 0)
+                        logger.info("Wikipedia Resource '%s" % rid)
+
+                        if not self.resource_exists(rid):
+                            logger.info("Could not find resource '%s" % rid)
```

Ruff (0.8.2)

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)